docs/doxygen/demos/http_demo.dox

```diff
@@ -0,0 +1,65 @@
+
+/**
+@page http_demo coreHTTP
+@brief These demos demonstrate usage of the coreHTTP library.
+
+@section http_demo_plaintext HTTP Plaintext Demo
+@brief A demo of the coreHTTP library that establishes a plaintext TCP connection with a server, sends HTTP requests, then logs their responses.
+
+<p>
+A POSIX socket-based transport interface implementation is used to establish a
+plaintext connection over port 80 to an HTTP server. However, if a connection 
+failure occurs, then the connection establishment is retried for @ref MAX_RETRY_ATTEMPTS.
+Once a connection is established, HTTP requests are sent for the following HTTP methods:
+`GET`, `HEAD`, `PUT`, and `POST`. The respective responses from each of these 
+requests are logged.
+</p>
+
+@image html http_demo_plaintext.png width=100%
+<div class="caption" style="text-align:center">
+HTTP Plaintext Demo Workflow — Note that these steps are repeated indefinitely
+</div>
+
+@section http_demo_basic_tls HTTP Basic TLS Demo
+@brief A demo of the coreHTTP library that establishes a TLS session on top of a TCP connection
```

Comment:
Maybe we can differentiate this description from the mutual auth one like the MQTT demo description does?
https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/master/docs/doxygen/demos/demos_subpages.dox#L3